File: wrappers/mvc/demos/Kendo.Mvc.Examples/Controllers/DataViz/Bar_Charts/Local_DataController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web.Mvc;
using Kendo.Mvc.Examples.Models;

namespace Kendo.Mvc.Examples.Controllers
{
    public partial class Bar_ChartsController : Controller
    {
        private const string HighlightColor = "#aaa";

        public ActionResult Local_Data()
        {
            var internetUsers = ChartDataRepository.InternetUsers();
            var viewModel = new List<BarChartsLocalDataViewModel>();

            for (var i = 1; i < internetUsers.Count; i++)
            {
                var data = internetUsers[i];
                var model = new BarChartsLocalDataViewModel(data);
                if (internetUsers[i - 1].Value > data.Value)
                {
                    // Highlight years with less users than the previous year
                    model.Color = HighlightColor;
                }

                viewModel.Add(model);
            }

            return View(viewModel);
        }
    }
}

File: wrappers/java/spring-demos/src/main/java/com/kendoui/spring/controllers/barcharts/LocalDataController.java
package com.kendoui.spring.controllers.barcharts;

import java.util.ArrayList;
import java.util.List;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

import com.kendoui.spring.models.ChartDataRepository;
import com.kendoui.spring.models.InternetUsers;
import com.kendoui.spring.models.BarChartsLocalDataViewModel;

@Controller("dataviz-bar_chart-local_data-controller")
@RequestMapping(value="/dataviz/bar-charts/")
public class LocalDataController {
    private final String HIGHLIGHT_COLOR = "#aaa";
    
    @RequestMapping(value = "/local-data", method = RequestMethod.GET)
    public String index(Model model) {
        List<InternetUsers> internetUsers = ChartDataRepository.InternetUsers();
        List<BarChartsLocalDataViewModel> viewModel =  new ArrayList<BarChartsLocalDataViewModel>();
        
        for (int i = 1; i < internetUsers.size(); i++)
        {
            InternetUsers data = internetUsers.get(i);
            BarChartsLocalDataViewModel item = new BarChartsLocalDataViewModel(data);
            if (internetUsers.get(i - 1).getValue() > data.getValue())
            {
                // Highlight years with less users than the previous year
                item.setColor(HIGHLIGHT_COLOR);
            }

            viewModel.add(item);
        }
        model.addAttribute("viewModel", viewModel);
     
        return "/dataviz/bar-charts/local-data";
    }
}
